### PR TITLE
Update to Checker Framework 3.14.0, and use shaded dataflow artifact that is specific to NullAway

### PR DESCRIPTION
**Pull request by mernst**
Currently, NullAway uses the "dataflow" Maven artifact:
https://mvnrepository.com/artifact/org.checkerframework/dataflow

NullAway conflicts with other tools that also use the `dataflow` artifact.  Therefore, the Checker Framework team has created a new shaded dataflow artifact just for NullAway to use:
https://mvnrepository.com/artifact/org.checkerframework/dataflow-nullaway

This pull request updates NullAway to use that artifact.

### CODE REVIEW

**Comment by msridhar** [comment]
Thanks for the contribution @mernst!  I had a WIP PR to update our Checker Framework dependence (#468); we will likely need to pull in changes from that PR to make this one pass tests / CI.  I will work on that soon.

**Comment by msridhar** [comment]
This change now looks good to me.  Before we land it, we should get a review from @lazaroclapp and also internal testing at Uber.

File: nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java:33]: Is this a check for an initializer block specifically? What happens to fields initialize with `T x = [expression]`? Wasn't that also a case we were covering by looking for `parameters == null`?

(I am probably wrong here, since I haven't thought about this code in a while, but want to make sure I understand the change)

File: nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayTryFinallyCases.java
Comment (lazaroclapp) [inline nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayTryFinallyCases.java:125]: 👍

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:228]: Was this method removed from `AnnotationUtils` and we are just inlining the implementation here? Or is there a different reason for this change?

**Review by lazaroclapp — COMMENTED** [review]
Looks good to me overall! 

Thanks so much for the new artifact, @mernst!

I have a few questions and, like @msridhar said, I'd want to try these changes internally before merging. Will likely be able to do so today or tomorrow. (Sorry for the late response, just got back from PTO)

File: nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java
Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java:33]: Here are the docs for the `ARBITRARY_CODE` case:

https://github.com/typetools/checker-framework/blob/de2f18948fb5ef6a4cf47163c8a6457404bc2cd7/dataflow/src/main/java/org/checkerframework/dataflow/cfg/UnderlyingAST.java#L24-L25

I _believe_ this case covers both initializer blocks and expressions.  @mernst do you know?  In any case all our tests from before still pass, and the check for `parameters == null` no longer works.

Comment (mernst) [reply nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java:33]: Yes, `ARBITRARY_CODE` is for initializer blocks.  The right-hand-sides of initializers will have an expression type; more relevantly, they don't have their own `UnderlyingAST`.

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java
Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:228]: The relevant methods from `AnnotationUtils` were deprecated.  The replacement APIs in `AnnotationUtils` take an `ExecutableElement` for the annotation rather than an annotation name, I think for efficiency.  We use these APIs from our contract-related handlers, and it would have taken a moderate refactoring (and possibly changes to the handler APIs) to be able to find and keep around the relevant `ExecutableElement`s.  Instead, I re-implemented these methods, under the assumption that if our performance was ok before, this should not cause a regression.

Let me know if you think we should document this better.

Comment (mernst) [reply nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:228]: You are right that the deprecation was because of performance.

**Review by lazaroclapp — APPROVED** [review]
Tested internally on one of our monorepos, no functional regression. There also doesn't seem to be a performance regression from this change vs current master, so let's go ahead and land.

However, in the process of checking this, I see a wall-clock duration regression of **3-4% of javac** for master vs 0.9.1 on a large target, after 10 runs, which worries me a bit. I am trying to see which of our recent commits might have introduced the issue. Might need to delta debug the commits between 0.9.1 and now.

**Comment by msridhar** [comment]
> However, in the process of checking this, I see a wall-clock duration regression of **3-4% of javac** for master vs 0.9.1 on a large target, after 10 runs, which worries me a bit. I am trying to see which of our recent commits might have introduced the issue. Might need to delta debug the commits between 0.9.1 and now.

So this regression is independent of this PR, if I understand correctly?  Would you like me to hold off on merging this PR until you track down that regression?

**Pull request merged**
